src/Microsoft.ML.Data/DataView/Transposer.cs

```diff
@@ -1559,58 +1534,26 @@ private void GetId(ref RowId id)
         }
 
         /// <summary>
-        /// This <see cref="ITransposeSchema"/> implementation wraps an <see cref="ISchema"/>,
+        /// This <see cref="ITransposeSlotTypeHolder"/> implementation wraps an <see cref="ISchema"/>,
         /// while indicating that no columns are actually transposed. This is useful for
         /// <see cref="ITransposeDataView"/> implementations that are wrapping a data view
         /// that might not implement that interface.
         /// </summary>
-        internal sealed class SimpleTransposeSchema : ITransposeSchema
+        internal sealed class SimpleTransposeSchema : ITransposeSlotTypeHolder
         {
             private readonly Schema _schema;
 
-            public int ColumnCount { get { return _schema.Count; } }
-
             public SimpleTransposeSchema(Schema schema)
             {
                 Contracts.CheckValue(schema, nameof(schema));
                 _schema = schema;
             }
 
-            public string GetColumnName(int col)
-            {
-                return _schema[col].Name;
-            }
-
-            public bool TryGetColumnIndex(string name, out int col)
-            {
-                return _schema.TryGetColumnIndex(name, out col);
-            }
-
-            public ColumnType GetColumnType(int col)
-            {
-                return _schema[col].Type;
-            }
-
             public VectorType GetSlotType(int col)
             {
-                Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
+                Contracts.CheckParam(0 <= col && col < _schema.Count, nameof(col));
                 return null;
             }
-
-            public IEnumerable<KeyValuePair<string, ColumnType>> GetMetadataTypes(int col)
-            {
-                return _schema[col].Metadata.Schema.Select(c => new KeyValuePair<string, ColumnType>(c.Name, c.Type));
-            }
-
-            public ColumnType GetMetadataTypeOrNull(string kind, int col)
-            {
-                return _schema[col].Metadata.Schema.GetColumnOrNull(kind)?.Type;
-            }
-
-            public void GetMetadata<TValue>(string kind, int col, ref TValue value)
-            {
-                _schema[col].Metadata.GetValue(kind, ref value);
-            }
         }
```

Comment:
This one is odd one.
Why I need schema object if all I care is schema.count?
Can you fix summary, it's no longer related to ISchema.
Maybe call it EmptySlotTypeHolder?

Reply:
Fixed.